api/v1alpha1/ipfscluster_types.go

```diff
@@ -60,28 +71,29 @@ type followParams struct {
 	Template string `json:"template"`
 }
 
-// networkConfig defines the configuration structure used for networking.
-type networkConfig struct {
+// NetworkConfig defines the configuration structure used for networking.
+type NetworkConfig struct {
+	// circuitRelays defines how many CircuitRelays should be created.
 	CircuitRelays int32 `json:"circuitRelays"`
+	// networkMode is a switch which defines whether this IPFSCluster will use
+	// the global IPFS network or create its own.
+	// +kubebuilder:validation:Enum={public,private}
+	NetworkMode NetworkMode `json:"networkMode,omitempty"`
 }
 
 // IpfsClusterSpec defines the desired state of the IpfsCluster.
 type IpfsClusterSpec struct {
-	// url defines the URL to be using as an ingress controller.
-	// +kubebuilder:validation:Optional
-	URL string `json:"url"`
```

Comment:
URL is deleted from spec because I did not see it used anywhere in the operator and I don't want to confuse users. Feel free to let me know if this needs to stay